app/src/main/java/com/github/se/polyfit/ui/screen/MapScreen.kt

```diff
@@ -24,88 +25,112 @@ import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.FlowPreview
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
 // Size in meters of the rayon of the circle to see other posts
-val MIN_SIZE_RAYON = 100f
-val MAX_SIZE_RAYON = 2000f
-val EPFL_LATITUDE = 46.5181
-val EPFL_LONGITUDE = 6.5659
+const val MIN_SIZE_RAYON = 100f
+const val MAX_SIZE_RAYON = 2000f
+const val EPFL_LATITUDE = 46.5181
+const val EPFL_LONGITUDE = 6.5659
+const val THREE_SECONDS = 3000L
 
+@OptIn(FlowPreview::class)
 @Composable
-fun MapScreen(
-    paddingValues: PaddingValues,
-    goToPost: () -> Unit,
-    mapviewMap: MapViewModel = hiltViewModel()
-) {
+fun MapScreen(paddingValues: PaddingValues, mapViewModel: MapViewModel = hiltViewModel()) {
 
   val posts = remember { mutableStateOf(listOf<Post>()) }
+
   val currentLocation = remember { mutableStateOf(Location.default()) }
-  val mutex = Mutex()
+
+  var selectedPost by remember { mutableStateOf<Post?>(null) }
 
   var isCircleShown by remember { mutableStateOf(false) }
 
+  var mainLaunchedIsFinished by remember { mutableStateOf(false) }
+
   val cameraPositionState = rememberCameraPositionState {
     position = CameraPosition.fromLatLngZoom(LatLng(EPFL_LATITUDE, EPFL_LONGITUDE), 15f)
   }
-  var sliderPosition by remember { mutableFloatStateOf((MIN_SIZE_RAYON + MAX_SIZE_RAYON) / 2) }
-
-  LaunchedEffect(currentLocation) {
-    mutex.withLock {
-      currentLocation.value = mapviewMap.getCurrentLocation()
-      mapviewMap.setLocation(currentLocation.value)
-    }
-
-    mapviewMap.listenToPosts()
-    mapviewMap.posts.observeForever { posts.value = it }
-  }
-  LaunchedEffect(sliderPosition) {
-    isCircleShown = true
-    mutex.withLock { mapviewMap.setRadius(sliderPosition.toDouble() / 1000) }
 
-    if (isCircleShown) {
-      mapviewMap.listenToPosts()
-      mapviewMap.posts.observeForever { posts.value = it }
-    }
-    delay(3000)
-    isCircleShown = false
+  var sliderPositionInMeters by remember {
+    mutableFloatStateOf((MIN_SIZE_RAYON + MAX_SIZE_RAYON) / 2)
   }
 
-  LaunchedEffect(key1 = mapviewMap) {
-    //        mapviewMap.getAllPost().collect { posts.value = it }
-
-    Log.i("Map", "MapScreen $posts")
-    mutex.withLock {
-      currentLocation.value = mapviewMap.getCurrentLocation()
-      mapviewMap.setLocation(currentLocation.value)
-    }
-    Log.i("Map", "currentLocation $currentLocation")
+  /**
+   * Effect to be launched when the mapViewModel changes. Initializes the current location and sets
+   * up the map and post listeners.
+   *
+   * This effect fetches the current location asynchronously, sets the location in the mapViewModel,
+   * and begins listening to posts. It also updates the camera position of the map and observes
+   * posts to update the state.
+   *
+   * @param mapViewModel The ViewModel that handles map-related data and operations.
+   */
+  LaunchedEffect(key1 = mapViewModel) {
```

Comment:
So, because every time the user navigates out of the map view and comes back, a new model will be injected, instead of using the mapViewModel, you could use a `Unit` to make the code a bit easier to understand:

```suggestion
LaunchedEffect(Unit) {

```

Reply:
I tried both, it doesn't change at all